lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -16,23 +26,101 @@ def actions(entity)
             actions << 'par' if can_ipo_any?(entity) || can_float_minor?(entity)
             actions << 'buy_company' if !purchasable_companies(entity).empty? || !buyable_bank_owned_companies(entity).empty?
             actions << 'sell_shares' if can_sell_any?(entity)
-
-            actions << 'pass' if !can_float_minor?(entity) && !bought? && !actions.empty?
+            actions << 'convert' if can_convert_any?
+            actions << 'pass' if !can_float_minor?(entity) && !actions.empty?
             actions
           end
 
+          def corporation_actions(entity)
+            return [] unless can_convert?(entity)
+
+            %w[convert pass]
+          end
+
           def can_buy_any_from_ipo?(entity)
             return unless @game.corporations.all?(&:ipoed)
 
             super
           end
 
+          def can_buy?(entity, bundle)
+            action = @round.current_actions.find { |a| a.is_a?(Engine::Action::Convert) }
+            return false if action && action&.entity != bundle.corporation
```

Comment:
You're now setting `@converted` in process_convert`, so can't you just check that value here, instead of digging through the action history?